plutus-playground-server/usecases/Game.hs

```diff
@@ -117,7 +117,7 @@ guess = do
     void (submitTxConstraintsSpending gameInstance unspentOutputs tx)
 
 game :: AsContractError e => Contract GameSchema e ()
-game = lock <|> guess
+game = lock `select` guess
```

Comment:
hm, there is an alternative instance for Eff if it has NonDet in it. I guess we can't reuse that... This is a downside of having only one type for effectful things, you can only have one instance for it!

Reply:
We could reuse it if we had an `Alternative` instance for `Contract` (because then we could just use `NonDet`), but I don't think we should have such an instance because there unit (`empty`) shouldn't be exposed to the user, because we don't want backtracking in the actual contracts - you can't just go back a couple of requests because we they can't be undone. They only way you can abort a branch is if you throw an error and hopefully catch it somewhere.

(There are rollbacks, which work a bit like backtracking, but those aren't controlled by the contract author either)

Reply:
Hm, do we want to restrict when people can catch errors too, then?